### PR TITLE
MM-27497 Patch localForage to fix logging out on old Edge

### PR DESCRIPTION
**Pull request by hmhealey**
When logging out, we call a method of `localForage` that in turn attempts to call a browser method that's not supported on pre-Chromium versions of Edge. There's a PR submitted to fix that (https://github.com/localForage/localForage/pull/971), but it's waiting to be merged and released, so I'm taking a page from the mobile app's handbook and using `patch-package` to temporarily apply the patch at install time since it's easier than maintaining a fork until the next version of `localForage` is released.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-27497

### CODE REVIEW

File: patches/localforage+1.7.4.patch [hunk 1]
```diff
@@ -0,0 +1,8390 @@
...
...object ArrayBuffer]"===c||a.buffer&&"[object ArrayBuffer]"===Ua.call(a.buffer))){var d,e=Ga;a instanceof ArrayBuffer?(d=a,e+=Ia):(d=a.buffer,"[object Int8Array]"===c?e+=Ka:"[object Uint8Array]"===c?e+=La:"[object Uint8ClampedArray]"===c?e+=Ma:"[object Int16Array]"===c?e+=Na:"[object Uint16Array]"===c?e+=Pa:"[object Int32Array]"===c?e+=Oa:"[object Uint32Array]"===c?e+=Qa:"[object Float32Array]"===c?e+=Ra:"[object Float64Array]"===c?e+=Sa:b(new Error("Failed to get type for BinaryArray"))),b(e+O(d))}else if("[object Blob]"===c){var f=new FileReader;f.onload=function(){var c=Ea+a.type+"~"+O(this.result);b(Ga+Ja+c)},f.readAsArrayBuffer(a)}else try{b(JSON.stringify(a))}catch(c){console.error("Couldn't convert value into a JSON string: ",a),b(null,c)}}function Q(a){if(a.substring(0,Ha)!==Ga)return JSON.parse(a);var b,c=a.substring(Ta),d=a.substring(Ha,Ta);if(d===Ja&&Fa.test(c)){var e=c.match(Fa);b=e[1],c=c.substring(e[0].length)}var f=N(c);switch(d){case Ia:return f;case Ja:return g([f],{type:b});case Ka:return new Int8Array(f);case La:return new Uint8Array(f);case Ma:return new Uint8ClampedArray(f);case Na:return new Int16Array(f);case Pa:return new Uint16Array(f);case Oa:return new Int
```
Comment (hmhealey) [inline patches/localforage+1.7.4.patch:5394]: These are the actual changes to fix the issue. The rest of the changes are because the library checks in their compiled files as well.

Comment (hmhealey) [inline patches/localforage+1.7.4.patch:5394]: These are the actual changes to fix the issue. The rest of the changes are because the library checks in their compiled files as well.

**Comment by hmhealey** [comment]
Lines 5373 to 5394 of the patch are the actual changes to fix the issue. The rest of the changes are because the library checks in their compiled files as well.

This link works, but you have to click to load the patch for it to scroll you to the right place.
https://github.com/mattermost/mattermost-webapp/pull/6308/files#diff-4499b04d71f37353f7e77b7b0cef7e29R5373-R5394

**Comment by hmhealey** [comment]
The new file was actually a problem. I copied files into node_modules wrong when preparing the patch. That should be fixed now, and the patch is much easier to read now.

**Comment by calebroseland** [comment]
Locally, I'm getting:

```
HTTP401: DENIED - The requested resource requires user authentication.
(Fetch)POST - http://mattermost.localdev:8065/api/v4/channels/members/me/view
```
```
Unhandled promise rejection TypeError: Invalid calling object  main.b3747573886923126066.js (239,43173)

[object Error]: {description: "Invalid calling object", message: "Invalid calling object", number: -2147418113, stack: "TypeError: Invalid calling object at Anonymous function (http://mattermost.localdev:8065/static/main.b3747573886923126066.js:199:48914) at T (http://mattermost.localdev:8065/static/main.b3747573886923126066.js:199:44009) at Anonymous function (http://mattermost.localdev:8065/static/main.b3747573886923126066.js:199:48814) at Anonymous function (http://mattermost.localdev:8065/static/main.b3747573886923126066.js:187:387048) at n (http://mattermost.localdev:8065/static/main.b3747573886923126066.js:239:42727)"}
```

**Comment by hmhealey** [comment]
Yeah, I'm seeing that as well. I'll take a look since clearly this fix isn't working as well as I'd hoped

**Comment by hmhealey** [comment]
I did some more testing locally, and for some reason the fix was still causing Edge to throw errors. I've instead gone with just downgrading `localforage` for the time being. Hopefully they'll have addressed the issue (or we'll have dropped old Edge :p) by the time we have to upgrade again.

**Review by jgilliam17 — APPROVED** [review]
Thank you @hmhealey 
Tested, looks good to merge.
- Verified user can logout successfully on the Microsoft Edge, returns to login screen. No console errors.
